Nercdl-330 notebook restyling

andrew-longmore-tessella opened the pull request:
Restyled to match design in NERCDL-330.

alex-lee-tessella commented on code/workspaces/web-app/src/components/stacks/StackCards.js:
```diff
@@ -5,11 +5,11 @@ import StackCard from './StackCard';
 const StackCards = ({ stacks, typeName, openStack, deleteStack, editStack, openCreationForm, userPermissions,
   createPermission, openPermission, deletePermission, editPermission }) => (
-  <Grid container spacing={2}>
+  <Grid container spacing={0}>
```
Not sure the Grid is necessary any more. Might be clearer what is going on if it was just wrapped in a div with
```
display: flex;
flex-direction: column;
```

andrew-longmore-tessella replied:
Initially kept it as a grid in case we wanted to revert, but agree best to change.  Thanks, done.

alex-lee-tessella commented on code/workspaces/web-app/src/components/stacks/StackStatus.js:
```diff
@@ -8,11 +8,10 @@ const { getStatusKeys, getStatusProps } = statusTypes;
-    paddingRight: theme.spacing(0.5),
-    paddingTop: theme.spacing(0.5),
-    paddingBottom: theme.spacing(0.25),
+    marginLeft: 4,
```
again might be worth replacing `4`s with `theme.spacing(1)`

andrew-longmore-tessella replied:
Thanks, done.

alex-lee-tessella commented:
> Looking good. I'm wondering whether we need to do something different with the status label. The sharp corners clash against the rounded corner buttons below for me.
> 
> Approved anyway as that could be tweaked in another PR if we want to.
> ![Screenshot 2019-09-04 at 13 47 57](https://user-images.githubusercontent.com/19606036/64256099-a4a6b980-cf1a-11e9-9a39-130a50d69c8b.png)

I agree with this. In the mock-ups they have a border radius the same as the buttons but Andy rightly pointed out that these then start looking like buttons so might need some more thought. 

Another point on this image, it would be nice if there was no space between the buttons. I think `Button` has a prop for making itself full width so could try those out on it to see if they then fill the space nicely.

alex-lee-tessella commented:
![image](https://user-images.githubusercontent.com/52826110/64258127-9064bb80-cf1e-11e9-836d-e3751205f67f.png)

![image](https://user-images.githubusercontent.com/52826110/64258943-06b5ed80-cf20-11e9-8a2d-b5c6f65ea1dd.png)

Spacing is a bit off in the row. In the mock-ups there is no margin on the sides which I think looks a bit cleaner and there is some more space between the logo and the title/description. There was also a border around the logos in the mock-ups which I think adds a bit of uniformity to the different logos so might be worth adding.

Vertical alignment is slightly off as well; think it just needs an `align-items: center` on the `.cardDiv` class. The fact it is wrapped in a Card also seems to be adding uneven spacing to the bottom of the card when it is the last-child which is strange.

Good work on making it handle resizing nicely.

![image](https://user-images.githubusercontent.com/52826110/64258757-b63e9000-cf1f-11e9-89b9-234a7b307fcf.png)

When there are no cards there is still a dividing line above the `add` button.

These things could be done in a future tweak as Gary suggested perhaps. Just wanted to make note of them so there is a record of anything that might need to be done in a future task.

andrew-longmore-tessella commented on code/workspaces/web-app/src/components/modal/CreateNotebookDialog.js:
```diff
@@ -4,26 +4,17 @@ import Dialog from '@material-ui/core/Dialog';
   <Dialog open={true} maxWidth="md">
-    <div style={{ margin: 10, display: 'flex', flexDirection: 'row' }}>
-      <div>
+    <div style={{ margin: 10 }}>
```
Thanks, done.

andrew-longmore-tessella commented on code/workspaces/web-app/src/components/stacks/NewStackButton.js:
```diff
@@ -12,11 +12,7 @@ const styles = theme => ({
-    display: 'flex',
-    alignItems: 'center',
-    justifyContent: 'left',
+    borderTop: '1px solid',
```
Thanks, done.

andrew-longmore-tessella commented on code/workspaces/web-app/src/components/stacks/StackCard.js:
```diff
@@ -48,29 +59,32 @@ function styles(theme) {
 const StackCard = ({ classes, stack, openStack, deleteStack, editStack, typeName, userPermissions,
   openPermission, deletePermission, editPermission }) => <Card className={classes.card}>
```
Thanks, done.

andrew-longmore-tessella commented on code/workspaces/web-app/src/components/stacks/StackCardActions.js:
```diff
@@ -9,10 +9,10 @@ const { READY } = statusTypes;
 const StackCardActions = ({ stack, openStack, deleteStack, editStack, userPermissions, openPermission,
   deletePermission, editPermission }) => (
-  <CardActions style={{ paddingLeft: 8, paddingRight: 8 }}>
+  <CardActions style={{ padding: 0, marginTop: 4, justifyContent: 'space-between' }}>
```
Thanks, done.

andrew-longmore-tessella commented:
Addressed most of these comments.  The status label is still unrounded to distinguish it from the buttons, since it's not clear how to proceed.  I've reduced the spacing between the buttons, it's now pretty much the min that you get with the flex div; and I've also done some work on the image border and general spacing.
